src/dvc_render/vega.py

```diff
@@ -117,22 +160,258 @@ def generate_markdown(self, report_path=None) -> str:
+        self, split_anchors: List[str], optional_anchors: List[str]
+    ):
+        self._fill_optional_anchor(split_anchors, optional_anchors, "group_by", ["rev"])
+        self._fill_optional_anchor(
+            split_anchors, optional_anchors, "pivot_field", "datum.rev"
+        )
+        for anchor in optional_anchors:
+            self.template.fill_anchor(anchor, {})
+
+    def _process_multi_source_plot(
+        self,
+        split_anchors: List[str],
+        optional_anchors: List[str],
+        y_definitions: List[Dict[str, str]],
+    ):
+        varied_keys, varied_values = self._collect_variations(y_definitions)
+        domain = self._get_domain(varied_keys, varied_values)
+
+        self._fill_optional_multi_source_anchors(
+            split_anchors, optional_anchors, varied_keys, domain
+        )
+        return varied_keys
+
+    def _collect_variations(
+        self, y_definitions: List[Dict[str, str]]
+    ) -> Tuple[List[str], Dict[str, set]]:
+        varied_values = defaultdict(set)
+        for defn in y_definitions:
+            for key in FILENAME_FIELD:
+                varied_values[key].add(defn.get(key, None))
+            varied_values[CONCAT_FIELDS].add(
+                FIELD_SEPARATOR.join([defn.get(FILENAME, ""), defn.get(FIELD, "")])
+            )
+
+        values_match_variations = []
+        less_values_than_variations = []
+
+        for filename_or_field in FILENAME_FIELD:
+            value_set = varied_values[filename_or_field]
+            num_values = len(value_set)
+            if num_values == 1:
+                continue
+            if num_values == len(y_definitions):
+                values_match_variations.append(filename_or_field)
+                continue
+            less_values_than_variations.append(filename_or_field)
+
+        if values_match_variations:
+            values_match_variations.extend(less_values_than_variations)
+            values_match_variations.sort(reverse=True)
+            return values_match_variations, varied_values
+
+        less_values_than_variations.sort(reverse=True)
+        return less_values_than_variations, varied_values
+
+    def _fill_optional_multi_source_anchors(
+        self,
+        split_anchors: List[str],
+        optional_anchors: List[str],
+        varied_keys: List[str],
+        domain: List[str],
+    ):
+        if not optional_anchors:
+            return
+
+        grouped_keys = ["rev", *varied_keys]
+        self._fill_optional_anchor(
+            split_anchors, optional_anchors, "group_by", grouped_keys
+        )
+        self._fill_optional_anchor(
+            split_anchors,
+            optional_anchors,
+            "pivot_field",
+            " + '::' + ".join([f"datum.{key}" for key in grouped_keys]),
+        )
+
+        concat_field = FIELD_SEPARATOR.join(varied_keys)
+        self._fill_optional_anchor(
+            split_anchors, optional_anchors, "row", {"field": concat_field}
```

Comment:
[C] when `row` is in the template and VS Code is the caller we may need to add the number of rows into the `_split_content` so we can adjust the plot's height.

Reply:
right now we will set to 300